controllers/capabilities/mapping_rules.go

```diff
@@ -182,15 +183,13 @@ func (t *ProductThreescaleReconciler) reconcileMappingRuleWithPosition(desired c
 	return nil
 }
 
-func (t *ProductThreescaleReconciler) createNewMappingRuleWithPosition(desired capabilitiesv1beta1.MappingRuleSpec, desiredPosition int) error {
-	metricID, err := t.productEntity.FindMethodMetricIDBySystemName(desired.MetricMethodRef)
+func (t *ProductThreescaleReconciler) createNewMappingRuleWithPosition(desired capabilitiesv1beta1.MappingRuleSpec, desiredPosition int, metricID int64) error {
+	isValidRule, err := t.validateMappingRulesDuplication(desired, metricID)
 	if err != nil {
-		return fmt.Errorf("Error creating product [%s] mappingrule: %w", t.resource.Spec.SystemName, err)
+		return err
 	}
-
-	if metricID < 0 {
-		// Should not happen as metric and method references have been validated and should exists
-		return errors.New("product metric method ref for mapping rule not found")
+	if !isValidRule {
```

Comment:
I'm trying to understand under what condition we would end up in this block.
The validateMappingRulesDuplication returns:
1. false and error - if there's an error in getting mapping rules - means we are not going to be here since we are going to exit above in the if err != nil
2. false and error if pattern is same, method is same and metric id is same - means we are not going to be here since we are going to exit above in the if err != nil
3. true and nil - we are not going to be here is isValidRule is true.

This checks from what it looks to me doesn't do anything?

Reply:
Thank you for comment. 
- if error happens in `validateMappingRulesDuplication()` , it can happen [here](https://github.com/valerymo/3scale-operator/blob/THREESCALE-10238/controllers/capabilities/mapping_rules.go#L219-L222)  . In this case we can't say nothing about Duplication as we didn't check it. 
- from my pov - we need have these 2 checks to provide correct log. But I improved error message,  and hope it's better now. Done in separate commit. Thanks for comment.